internal/compiler-interface/src/main/java/xsbti/compile/CachedCompilerProvider.java

```diff
@@ -22,5 +22,5 @@
 	ScalaInstance scalaInstance();
 
 	/** Return a new cached compiler from the usual compiler input. */
-	CachedCompiler newCachedCompiler(String[] arguments, Output output, Logger log, Reporter reporter);
+	CachedCompiler2 newCachedCompiler(String[] arguments, Output output, Logger log, Reporter reporter);
```

Comment:
AnalysisCompiler should be able to return either Dotty CachedCompiler or Scala 2.x CachedCompiler2, so if you want to split them, this should return AnyRef.